src/Calculator/BcMathCalculator.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * This file is part of the Money library
+ *
+ * Copyright (c) 2011-2013 Mathias Verraes
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Money\Calculator;
+
+use InvalidArgumentException;
+use Money\CalculatorInterface;
+use Money\Money;
+use Money\Number;
+
+/**
+ * Class BcMathCalculator
+ * @package Money\Calculator
+ */
+final class BcMathCalculator implements CalculatorInterface
+{
+
+    /**
+     * @var string
+     */
+    private $scale;
+
+    /**
+     * BcMathCalculator constructor.
+     * @param int $scale
+     */
+    public function __construct($scale = 14)
+    {
+        $this->scale = $scale;
+    }
+
+
+    /**
+     * @return bool
+     */
+    public function supported()
+    {
+        return extension_loaded('bcmath');
+    }
+
+    /**
+     * @param $a
+     * @param $b
+     * @return int
+     */
+    public function compare($a, $b)
+    {
+        return bccomp($a, $b);
+    }
+
+    /**
+     * @param $amount
+     * @param $addend
+     * @return mixed
+     */
+    public function add($amount, $addend)
+    {
+        return bcadd($amount, $addend);
+    }
+
+    /**
+     * @param $amount
+     * @param $subtrahend
+     * @return mixed
+     */
+    public function subtract($amount, $subtrahend)
+    {
+        return bcsub($amount, $subtrahend);
+    }
+
+    /**
+     * @param $amount
+     * @param $multiplier
+     * @return mixed
+     */
+    public function multiply($amount, $multiplier)
+    {
+        return bcmul($amount, $multiplier, $this->scale);
+    }
+
+    /**
+     * @param $amount
+     * @param $divisor
+     * @return mixed
+     */
+    public function divide($amount, $divisor)
+    {
+        return bcdiv($amount, $divisor, $this->scale);
+    }
+
+    /**
+     * @param $number
+     * @return mixed
+     */
+    public function ceil($number)
+    {
+        $decimal_separator_position = strpos($number, '.');
```

Comment:
Please use camelCase variables and add a new line after this statement to increase readability.